src/routes/PlayQuiz/index.tsx

```diff
@@ -151,17 +184,98 @@ export default function PlayQuiz() {
     }
   }
 
+  function getAllQuestions(categories) {
+    return categories.reduce((acc, category) => acc.concat(category.questions), [] as IQuestion[]);
+  }
+
   function showQuestion(questionId: string | number, categoryId: string | number) {
-    if (categoryId) {
-      const question = categoriesInfo[categoryId].questions.find((q) => q.questionId === questionId);
+    const category = quizInfo.categories.find((x) => x.categoryId === categoryId);
+
+    if (category) {
+      const question = category.questions.find((q) => q.questionId === questionId);
 
       setSelectedQuestion(question || null);
       setIsPlaying(true);
     }
   }
 
   function shouldShowTimer() {
-    return ((!selectedQuestion && !!selectionTimeLimit) || (selectedQuestion && showQuestionTimer)) && !winner;
+    return (
+      ((!selectedQuestion && !!selectionTimeLimit) || (selectedQuestion && showQuestionTimer)) &&
+      !winner &&
+      isGameStarted
+    );
+  }
+
+  function getSelectedOptionId(selectedQuestion) {
+    const selectedOptionData = selectedOptionsData.find((x) => x.questionId === selectedQuestion.questionId);
+
+    if (selectedOptionData) {
+      return selectedOptionData.selectedOptionId;
+    } else {
+      return '';
+    }
+  }
+
+  function getQuestionColor(question: IQuestion): string {
+    if (attemptedQuestionIds.includes(question.questionId)) {
+      const correctOptionId = question.options.find((x) => x.isCorrect)?.optionId;
+      const selectedOptionId = selectedOptionsData.find((x) => x.questionId === question.questionId)?.selectedOptionId;
+
+      return correctOptionId === parseInt(selectedOptionId) ? 'green' : 'red';
+    } else {
+      return 'blue';
+    }
+  }
+
+  function getNameInitials(name) {
+    const arr = name.split(' ');
+
+    return arr[0][0].toUpperCase() + (arr[1] ? arr[1][0].toUpperCase() : arr[0][1].toUpperCase());
+  }
+
+  function getTeamAvatar(question) {
```

Comment:
This could be a separate component where we pass question & it renders an avatar